### PR TITLE
NP-958:service-first-sketch

### PR DESCRIPTION
**Pull request by axthosarouris**
### Description:
This PR contains: 

* A first sketch of how I am planning to implement the DatabaseService
* An approach of storing the data in the database and creating DTOs to communitcate with the rest   of the backend or clients.
* Unit tests (of course)
* Feature files and feature tests. All functionality and (desired functionality) of the service  

### What the PR is missing (and will be done in next PRs):

1. miscellaneous gradle plugins
2.  test coverage
3.  Lambda handlers
4. The feature tests should be against the lambda handlers and not the database service.

### Goals of this PR:
1. Do you agree with the structure?
2. Comments about the approach on the DatabaseService the database and DTO objects and the feature tests

### CODE REVIEW

File: .aws-sam/template.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  UserRoleMappings:
+    Type: AWS::DynamoDB::Table
+    Properties:
+      TableName: UsersRoles
+      AttributeDefinitions:
+      - AttributeName: EntityId
+        AttributeType: S
+      - AttributeName: Type
+        AttributeType: S
+      BillingMode: PROVISIONED
```
Comment (janbj) [inline .aws-sam/template.yaml:14]: Maybe use PAY_PER_REQUEST

File: src/main/java/no/unit/nva/handlers/GetUserHandler.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.amazonaws.services.lambda.runtime.Context;
+import no.unit.nva.database.UserDb;
+import nva.commons.exceptions.ApiGatewayException;
+import nva.commons.handlers.ApiGatewayHandler;
+import nva.commons.handlers.RequestInfo;
+import nva.commons.utils.Environment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GetUserHandler extends ApiGatewayHandler<Void, UserDb> {
```
Comment (janbj) [inline src/main/java/no/unit/nva/handlers/GetUserHandler.java:12]: Do you intend to have more than one Lambda in this project?

Comment (axthosarouris) [reply src/main/java/no/unit/nva/handlers/GetUserHandler.java:12]: Yes. GetUser, AddUser, UpdateUser, GetRole, and maybe AddRole and UpdateRole

File: .aws-sam/template.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: A small description of the template
```
Comment (brinxmat) [inline .aws-sam/template.yaml:3]: A small description would be nice :)

File: .aws-sam/template.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+Resources:
+  UserRoleMappings:
+    Type: AWS::DynamoDB::Table
+    Properties:
+      TableName: UsersRoles
+      AttributeDefinitions:
+      - AttributeName: EntityId
```
Comment (brinxmat) [inline .aws-sam/template.yaml:10]: Does this define the structure?

File: src/main/java/no/unit/nva/database/UserDb.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    private String id;
+
+    @DynamoDBRangeKey(attributeName = "Type")
+    private String type;
+
+    @DynamoDBAttribute(attributeName = "name")
+    private String username;
+
+    @DynamoDBAttribute(attributeName = "roles")
+    private String roles;
```
Comment (brinxmat) [inline src/main/java/no/unit/nva/database/UserDb.java:25]: The naming implies that this is a list

File: src/main/java/no/unit/nva/database/UserDb.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import java.util.Objects;
+
+@DynamoDBTable(tableName = "UsersRoles")
+public class UserDb {
+
+    public static final String ROLE_DELIMITER = ",";
+    public static String TYPE = "User";
+
```
Comment (brinxmat) [inline src/main/java/no/unit/nva/database/UserDb.java:14]: Do we want to know the modified/created/etc. of this?

Comment (brinxmat) [reply src/main/java/no/unit/nva/database/UserDb.java:14]: Possibly not relevant right here, but we should also consider if changing this results in some cloudwatch event that propagated to CloudTrail so we know how the changes were effected.

**Review by brinxmat — APPROVED** [review]

**Pull request merged**